src/votekit/election_types.py

```diff
@@ -1,117 +1,140 @@
+        remaining: list[str] = self.election_state.remaining
+        ballots: list[Ballot] = self.election_state.profile.get_ballots()
+        fp_votes: dict
+        fp_order: list[str]
+        fp_votes, fp_order = compute_votes(remaining, ballots)
+        elected = []
+        eliminated = []
+
+        # if number of remaining candidates equals number of remaining seats, everyone is elected
+        if len(remaining) == self.seats - len(self.election_state.get_all_winners()):
+            elected = fp_order
+            remaining = []
+            ballots = []
+            # TODO: sort remaining candidates by vote share
+
+        # elect all candidates who crossed threshold
+        elif fp_votes[fp_order[0]] >= self.threshold:
+            for candidate in fp_order:
+                if fp_votes[candidate] >= self.threshold:
+                    elected.append(candidate)
+                    remaining.remove(candidate)
+                    ballots = self.transfer(
+                        candidate, ballots, fp_votes, self.threshold
+                    )
+        # since no one has crossed threshold, eliminate one of the people
+        # with least first place votes
+        elif self.next_round():
             lp_votes = min(fp_votes.values())
             lp_candidates = [
-                candidate for candidate, votes in fp_votes.items() if votes == lp_votes
+                candidate for candidate in fp_order if fp_votes[candidate] == lp_votes
             ]
             # is this how to break ties, can be different based on locality
             lp_cand = random.choice(lp_candidates)
+            eliminated.append(lp_cand)
             ballots = remove_cand(lp_cand, ballots)
-            candidates.remove(lp_cand)
-            self.eliminated.add(lp_cand)
-
-        return PreferenceProfile(ballots=ballots), Outcome(
-            elected=self.elected,
-            eliminated=self.eliminated,
-            remaining=set(candidates),
-            votes=fp_votes,
+            remaining.remove(lp_cand)
+
+        self.election_state = ElectionState(
+            curr_round=self.election_state.curr_round + 1,
+            elected=elected,
+            eliminated=eliminated,
+            remaining=remaining,
+            profile=PreferenceProfile(ballots=ballots),
+            previous=self.election_state,
         )
+        return self.election_state
 
-    def run_election(self) -> Outcome:
+    def run_election(self) -> ElectionState:
         """
         Runs complete STV election
         """
-        profile = deepcopy(self.profile)
-
         if not self.next_round():
             raise ValueError(
                 f"Length of elected set equal to number of seats ({self.seats})"
             )
 
         while self.next_round():
-            profile, outcome = self.run_step(profile)
+            self.run_step()
+
+        return self.election_state
 
-        return outcome
+    def get_init_profile(self):
```

Comment:
if the profile is accessible, any reason to use this function instead of accessing it as self.profile? (or self.initial_profile if you want to be more clear about its purpose)